### DISCUSSION TITLE
[API] deleting connections doesn't work

### DISCUSSION THREAD

**Discussion by NachoAlasia** [airbytehq/airbyte#33069] [Connector Questions]
Hi there!

Im using the API for deleting connections, following the docs
(https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/connections/delete,
https://reference.airbyte.com/reference/deleteconnection
Yes, I tried with this 2 options :sonrisa: )
but when I tried, response this.
```html
<html>
<head>
    <title>405 Not Allowed</title>
</head>
<script language="javascript" \>
    window.TRACKING_STRATEGY = "segment";
                window.AIRBYTE_VERSION = "0.40.26";
                window.API_URL = "/api/v1/";
                window.CONNECTOR_BUILDER_API_URL = "/connector-builder-api";
</script>.
```
Why this happen? If I use the API for the rest of  actions and work fine.
Why Im not allowed?
Btw, I tried this code in 2 different versions of Airbyte. (0.40.26) and (0.50.29)

**Comment by pgrant87** [comment]
Hi Nacho,

What does your delete request look like at the moment? If you can share this, I'll investigate further

A 405 error usually happens when a client attempts to use an HTTP method that is not allowed by the server. For example, this can occur when a user tries to use a method such as PUT or DELETE on a resource that only allows GET or POST requests.

I'd also advise trying this with the latest version of Airbyte to see if the same error is returned

**Reply by NachoAlasia** [reply]
Yes, I know that! for that its weird, I think Im doing well, but Im human :D,
Yes, for sure, I attached the 2 ways I tried.
![post-delete](https://github.com/airbytehq/airbyte/assets/152916929/72bc18a4-5af6-4b4c-baf7-20e7df547cec)
![delete-delete](https://github.com/airbytehq/airbyte/assets/152916929/9a1e9fd1-8767-4563-a4d6-5944639846e4)

Hope this pictures helps you.

I send this following this two docs.
https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/connections/delete,
https://reference.airbyte.com/reference/deleteconnection

Thanks!

**Comment by pgrant87** [comment]
Nacho can I ask what interface you're using here to send your request to the API? What authentication are you using?

If you use a simple `curl` are you able to delete a connection?

I've tried this locally on the latest version of Airbyte and was able to delete a test connection of mines, like so:

```
curl -s --location --request DELETE "http://localhost:8006/v1/connections/87bc5f93-af80-48d9-b181-58b3019fc93b" --header "Authorization: Basic YWlyYnl0ZTpwYXNzd29yZA=="
```

Thanks!

**Reply by NachoAlasia** [reply]
Yes, indeed, Im using Postman in this case, but I also try connecting with python. 
Im using python for automation and work well.
The authentication Im using its the same I use for the rest of the tasks, like create a source, sync a connector, etc,
Just for know, can I ask in wich version do you release this option (delete connections)? in the lastest or in other oldest?

**Reply by pgrant87** [reply]
I tried locally with Postman and again I'm able to delete the connection. One thing I've noticed that's different in your request is the port, have you changed this to 8000 or is this a typo? By default, it should be 8006. Can I also ask how you're getting your connection-id in this case and is it valid?

I can't find the exact version where the delete connection operation was released in the API from the changelog but it's fairly standard functionality so I'd assume it's been around from the start

**Reply by NachoAlasia** [reply]
Hi Grant!

Oh, I see! that's very weird! they develop this on EC2 and assign in this port (8000), idk why, but, As I was saying, the rest of the operations work fine,I don't think that's the problem .
 I tried with the CURL you gave me and didnt work either.
 Yes! for the connections ID I use 2 options, the first one, you can take it from the url connection it self, or I usually use the following doc: https://reference.airbyte.com/reference/listconnections.
 
 Btw, I tried changing the port and failed too.

**Reply by pgrant87** [reply]
Getting the `curl` to work is imperative as I would see that as the simplest way to use the API. In my curl command in the previous comment, I've passed my username and password encoded as base64 in the `--header` section as authentication. What type of authentication are you using when trying to hit the API with a curl?

Can I ask, what sort of Airbyte installation is this, is it on docker-compose or Kubernetes?

Would you be able to share with me a curl to the API that works alongside the one to the delete endpoint to see if there's any discrepancy I can see?

Thank you!

**Reply by NachoAlasia** [reply]
Im sending this curl, with authorization in base64 aswell.

curl -s --location --request DELETE http://airbyte-ec2.companyURL.com:8000/v1/connections/2841ed89-123123-adsas-37ebe0b671a8 --header "Authorization: Basic YW--------------E5elM="

This instalation is on docker-compose, deploying in a EC2

**Reply by NachoAlasia** [reply]
But if I tried this one, work!

curl -s --location --request POST http://airbyte-ec2companyURLl.com:8000/api/v1/workspaces/list --header "Authorization: Basic YW--------------E5elM="

There's the weird thing, I tried with the connection paused and unpaused.

**Reply by pgrant87** [reply]
Thanks for those, I appreciate it. Your request looks fine to me here.

The only other thing I'd like to check at this stage is the connectionID that you're passing. If you hit the connection endpoint with the following request, do you see the ID you're trying to delete listed there:

```
curl -s --location --request GET "http://<yourAirbyteUrl>:8000/v1/connections" --header "Authorization: Basic YWlyYnl0ZTpwYXNzd29yZA=="
```

**Reply by pgrant87** [reply]
Having a search on your 405 error, I also found the following: https://discuss.airbyte.io/t/starting-the-airbyte-api-on-a-self-hosted-instance/2698

This looks like an older version of the API but here the user had to pass `Content-Type` and `Accept json` as headers, this may be worth a try here too
![image](https://github.com/airbytehq/airbyte/assets/32844876/13c66cb4-78de-44b3-82a8-d3d6f47de30f)

**Comment by NachoAlasia** [comment]
I tried and this happend

![image](https://github.com/airbytehq/airbyte/assets/152916929/77c6a811-42c8-426e-9778-de1423b912a8)

Really, Idk whats going on here, its super weird

**Reply by pgrant87** [reply]
Thanks for trying that Nacho, would it be possible to share your `docker-compose.yaml` and your `.env` file to check on your configuration here? Please be sure to redact any sensitive information from these before sharing.

Thank you!

**Comment by andyWang1688** [comment]
hi I also have this issue, but if I send other requests, they will return the HTML code to me
![image](https://github.com/airbytehq/airbyte/assets/45682184/6075827e-0f06-4f95-945f-bbd408bfb35a)

**Reply by pgrant87** [reply]
Hi @1043038630 Can you confirm what version of Airbyte you're seeing the issue in? Do you have a docker or k8s install? Could you also provide the full delete request that fails? Thank you!

[a comment by an automated account is not shown]
